[proj4]Upgrade version to 6.1.1 and remove useless patches.

JackBoosY opened the pull request:
1. Upgrade proj4 to v6.1.1 
2. Change the source to github.
3. Add a new version of the dependency port `sqlite3[tool]`.
4. Remove useless patches.
5. According to the official [response](https://github.com/OSGeo/PROJ/issues/1479#issuecomment-515925289), change the generated library name to `proj.lib`.
6. Feature `database` need to install sqlite3:x86-windows/sqlite3:x64-windows on triplet arm/arm64/uwp.
7. Fix the default library name of proj4 used by gdal.

Related: #3258 #3408 #6180 #6949..

Commit:
[proj4]Upgrade version to 6.1.1 and remove useless patches.

Commit:
[libgeotiff]Fix build errors using proj4 6.1.1.

Commit:
[libspatialite]Fix build errors using proj4 6.1.1.

Commit:
[gdal]Fix build errors using proj4 6.1.1.

Commit:
[spatialite-tools]Fix build errors using proj4 6.1.1.

Commit:
[spatialite-tools]Fix build errors using proj4 6.1.1.

Commit:
[libgeotiff]Re-fix build errors using proj4 6.1.1.

Commit:
[vtk]Fix build errors using proj4 6.1.1.

Commit:
[proj4]Fix find sqlite binary failure.

Commit:
[proj4]Disable build tools when building proj4 in arm/uwp.

Commit:
[sqlite3/libgeotiff]Fix libdl library linkage. Re-fix use proj issue.

JackBoosY committed:
Merge branch 'master' into dev/jack/upgrade_proj4_6.1.1

dan-shaw committed:
add back dl libs

Rastaban committed:
Merge branch 'master' into dev/jack/upgrade_proj4_6.1.1

Commit:
[gdal]Fix dependent port proj library name.

Commit:
Merge branch 'dev/jack/upgrade_proj4_6.1.1' of https://github.com/JackBoosY/vcpkg into dev/jack/upgrade_proj4_6.1.1

Commit:
Merge branch 'master' of https://github.com/microsoft/vcpkg into dev/jack/upgrade_proj4_6.1.1

Commit:
[proj4]Set Windows library name to "proj" and revert fix-proj-name changes.

Commit:
[liblas]Fix build errors using proj4 6.1.1.

Commit:
[liblas]Fix build errors using proj4 6.1.1.

Commit:
[liblas]Fix build errors using proj4 6.1.1.

Commit:
[proj4]Fix feature database on triplet arm/uwp.

JackBoosY commented:
/azp run

Commit:
Merge branch 'master' of https://github.com/Microsoft/vcpkg into dev/jack/upgrade_proj4_6.1.1

JackBoosY commented:
/azp run

JackBoosY commented:
/azp run

Commit:
[gdal]Remove useless flag PROJ_STATIC.

Commit:
[proj4]Fix build error on x64-windows-static.

Neumann-A commented on ports/liblas/portfile.cmake:
```diff
@@ -15,6 +15,18 @@ vcpkg_extract_source_archive_ex(
     PATCHES fix-BuildError.patch
 )
 
+if (VCPKG_TARGET_IS_WINDOWS)
```
find_library actually works now in portfiles. so you could use something like:

```
find_library(SQLITE_RELEASE NAMES sqlite3 PATHS "${CURRENT_INSTALLED_DIR}/lib" NO_DEFAULT_PATH) # Depends on openssl and zlib(linux)
find_library(SQLITE_DEBUG NAMES sqlite3 sqlite3d PATHS "${CURRENT_INSTALLED_DIR}/debug/lib" NO_DEFAULT_PATH)
```

Commit:
[liblas]Use find_library to get proj4/sqlite3 libraries.

JackBoosY committed:
Merge branch 'master' into dev/jack/upgrade_proj4_6.1.1

ras0219-msft commented on ports/gdal/portfile.cmake:
```diff
@@ -252,6 +252,7 @@ if (NOT VCPKG_CMAKE_SYSTEM_NAME OR VCPKG_CMAKE_SYSTEM_NAME STREQUAL "WindowsStor
     # Release build
     ################
     message(STATUS "Building ${TARGET_TRIPLET}-rel")
+    message(STATUS "Release Options: ${NMAKE_OPTIONS_REL}")
```
Leftover debug messages

ras0219-msft commented on ports/liblas/fix-BuildError.patch:
```diff
@@ -41,7 +41,7 @@ index e35de06..f7c25d2 100644
 -  ${Boost_LIBRARIES})
 +  ${Boost_LIBRARIES}
-+  proj)
++  ${PROJ4_LIBRARIES} ${SQLITE3_LIBRARIES})
```
Why does this suddenly need sqlite3?

If this is a transitive dependency due to proj4, we should push this into the proj4 config files.

ras0219-msft commented on ports/sqlite3/CMakeLists.txt:
```diff
@@ -27,7 +27,7 @@ target_compile_definitions(
 if(NOT WIN32)
     find_package(Threads REQUIRED)
-    target_link_libraries(sqlite3 PRIVATE Threads::Threads ${CMAKE_DL_LIBS})
+    target_link_libraries(sqlite3 PRIVATE Threads::Threads PUBLIC ${CMAKE_DL_LIBS})
```
Sqlite doesn't use DL_LIBS in its public interface -- someone else in the chain must be causing any problems that this fixes.

Commit:
[proj4]Fix sqlite dependency.

JackBoosY commented:
@ras0219-msft Thanks for review.

Commit:
[proj4]Fix arm/uwp dependency.

Commit:
[proj4]Fix linux build: configure error.

Commit:
[proj4]Disable export proj4 namespace.

JackBoosY commented:
`otl`, `hungarian` and `libxmp-lite` fails to download its source.
Seems CI machine cannot connect to source website.

LLVM has copy file error,  I cannot repro it on my machine.

Commit:
[liblas]Delete proj dependency.

JackBoosY commented:
Only have OSX failure here:
```
FAILED: bin/Debug/liblas.2.4.0.dylib 
: && /Library/Developer/CommandLineTools/usr/bin/c++ -fPIC  -Wextra -Wall -Wno-unused-parameter -Wno-unused-variable -Wpointer-arith -Wcast-align -Wcast-qual -Wfloat-equal -Wredundant-decls -Wno-long-long -g  -dynamiclib -Wl,-headerpad_max_install_names -compatibility_version 3.0.0 -current_version 2.4.0 -o bin/Debug/liblas.2.4.0.dylib -install_name @rpath/liblas.3.dylib src/CMakeFiles/las.dir/detail/index/indexcell.cpp.o src/CMakeFiles/las.dir/detail/index/indexoutput.cpp.o src/CMakeFiles/las.dir/chipper.cpp.o src/CMakeFiles/las.dir/factory.cpp.o src/CMakeFiles/las.dir/classification.cpp.o src/CMakeFiles/las.dir/color.cpp.o src/CMakeFiles/las.dir/dimension.cpp.o src/CMakeFiles/las.dir/error.cpp.o src/CMakeFiles/las.dir/filter.cpp.o src/CMakeFiles/las.dir/header.cpp.o src/CMakeFiles/las.dir/index.cpp.o src/CMakeFiles/las.dir/point.cpp.o src/CMakeFiles/las.dir/reader.cpp.o src/CMakeFiles/las.dir/spatialreference.cpp.o src/CMakeFiles/las.dir/schema.cpp.o src/CMakeFiles/las.dir/transform.cpp.o src/CMakeFiles/las.dir/utility.cpp.o src/CMakeFiles/las.dir/variablerecord.cpp.o src/CMakeFiles/las.dir/writer.cpp.o src/CMakeFiles/las.dir/version.cpp.o src/CMakeFiles/las.dir/detail/utility.cpp.o src/CMakeFiles/las.dir/detail/sha1.cpp.o src/CMakeFiles/las.dir/detail/zippoint.cpp.o src/CMakeFiles/las.dir/detail/reader/header.cpp.o src/CMakeFiles/las.dir/detail/reader/reader.cpp.o src/CMakeFiles/las.dir/detail/reader/zipreader.cpp.o src/CMakeFiles/las.dir/detail/reader/cachedreader.cpp.o src/CMakeFiles/las.dir/detail/writer/header.cpp.o src/CMakeFiles/las.dir/detail/writer/point.cpp.o src/CMakeFiles/las.dir/detail/writer/zipwriter.cpp.o src/CMakeFiles/las.dir/detail/writer/writer.cpp.o -L/Users/vagrant/azure_agent/_work/3/s/installed/x64-osx/debug/lib /Users/vagrant/azure_agent/_work/3/s/installed/x64-osx/debug/lib/libtiffd.a /Users/vagrant/azure_agent/_work/3/s/installed/x64-osx/debug/lib/liblzmad.a /Users/vagrant/azure_agent/_work/3/s/installed/x64-osx/debug/lib/libjpeg.a /Users/vagrant/azure_agent/_work/3/s/installed/x64-osx/debug/lib/libz.a -lm /Users/vagrant/azure_agent/_work/3/s/installed/x64-osx/debug/lib/libgeotiff.a /Users/vagrant/azure_agent/_work/3/s/installed/x64-osx/debug/lib/libboost_program_options.a /Users/vagrant/azure_agent/_work/3/s/installed/x64-osx/debug/lib/libboost_thread.a /Users/vagrant/azure_agent/_work/3/s/installed/x64-osx/debug/lib/libboost_system.a /Users/vagrant/azure_agent/_work/3/s/installed/x64-osx/debug/lib/libboost_iostreams.a /Users/vagrant/azure_agent/_work/3/s/installed/x64-osx/debug/lib/libboost_filesystem.a /Users/vagrant/azure_agent/_work/3/s/installed/x64-osx/debug/lib/libboost_chrono.a /Users/vagrant/azure_agent/_work/3/s/installed/x64-osx/debug/lib/libboost_date_time.a /Users/vagrant/azure_agent/_work/3/s/installed/x64-osx/debug/lib/libboost_atomic.a /Users/vagrant/azure_agent/_work/3/s/installed/x64-osx/debug/lib/libboost_regex.a /Users/vagrant/azure_agent/_work/3/s/installed/x64-osx/debug/lib/libtiffd.a /Users/vagrant/azure_agent/_work/3/s/installed/x64-osx/debug/lib/liblzmad.a -lproj /Users/vagrant/azure_agent/_work/3/s/installed/x64-osx/debug/lib/libjpeg.a /Users/vagrant/azure_agent/_work/3/s/installed/x64-osx/debug/lib/libz.a -lm && :
Undefined symbols for architecture x86_64:
  "_sqlite3_bind_double", referenced from:
      osgeo::proj::io::DatabaseContext::Private::run(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::list<osgeo::proj::io::SQLValues, std::__1::allocator<osgeo::proj::io::SQLValues> > const&) in libproj.a(factory.cpp.o)
  "_sqlite3_bind_text", referenced from:
      osgeo::proj::io::DatabaseContext::Private::run(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::list<osgeo::proj::io::SQLValues, std::__1::allocator<osgeo::proj::io::SQLValues> > const&) in libproj.a(factory.cpp.o)
  "_sqlite3_close", referenced from:
      osgeo::proj::io::DatabaseContext::Private::closeDB() in libproj.a(factory.cpp.o)
  "_sqlite3_column_count", referenced from:
      osgeo::proj::io::DatabaseContext::Private::run(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::list<osgeo::proj::io::SQLValues, std::__1::allocator<osgeo::proj::io::SQLValues> > const&) in libproj.a(factory.cpp.o)
  "_sqlite3_column_text", referenced from:
      osgeo::proj::io::DatabaseContext::Private::run(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::list<osgeo::proj::io::SQLValues, std::__1::allocator<osgeo::proj::io::SQLValues> > const&) in libproj.a(factory.cpp.o)
  "_sqlite3_create_function", referenced from:
      osgeo::proj::io::DatabaseContext::Private::registerFunctions() in libproj.a(factory.cpp.o)
  "_sqlite3_errmsg", referenced from:
      osgeo::proj::io::DatabaseContext::Private::run(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::list<osgeo::proj::io::SQLValues, std::__1::allocator<osgeo::proj::io::SQLValues> > const&) in libproj.a(factory.cpp.o)
  "_sqlite3_finalize", referenced from:
      osgeo::proj::io::DatabaseContext::Private::closeDB() in libproj.a(factory.cpp.o)
  "_sqlite3_open_v2", referenced from:
      osgeo::proj::io::DatabaseContext::Private::open(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, projCtx_t*) in libproj.a(factory.cpp.o)
      osgeo::proj::io::DatabaseContext::Private::attachExtraDatabases(std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&) in libproj.a(factory.cpp.o)
  "_sqlite3_prepare_v2", referenced from:
      osgeo::proj::io::DatabaseContext::Private::run(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::list<osgeo::proj::io::SQLValues, std::__1::allocator<osgeo::proj::io::SQLValues> > const&) in libproj.a(factory.cpp.o)
  "_sqlite3_reset", referenced from:
      osgeo::proj::io::DatabaseContext::Private::run(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::list<osgeo::proj::io::SQLValues, std::__1::allocator<osgeo::proj::io::SQLValues> > const&) in libproj.a(factory.cpp.o)
  "_sqlite3_result_double", referenced from:
      osgeo::proj::io::PROJ_SQLITE_pseudo_area_from_swne(sqlite3_context*, int, sqlite3_value**) in libproj.a(factory.cpp.o)
  "_sqlite3_result_int", referenced from:
      osgeo::proj::io::PROJ_SQLITE_intersects_bbox(sqlite3_context*, int, sqlite3_value**) in libproj.a(factory.cpp.o)
  "_sqlite3_result_null", referenced from:
      osgeo::proj::io::PROJ_SQLITE_pseudo_area_from_swne(sqlite3_context*, int, sqlite3_value**) in libproj.a(factory.cpp.o)
      osgeo::proj::io::PROJ_SQLITE_intersects_bbox(sqlite3_context*, int, sqlite3_value**) in libproj.a(factory.cpp.o)
  "_sqlite3_snprintf", referenced from:
      osgeo::proj::internal::toString(int) in libproj.a(internal.cpp.o)
      osgeo::proj::internal::toString(double, int) in libproj.a(internal.cpp.o)
  "_sqlite3_step", referenced from:
      osgeo::proj::io::DatabaseContext::Private::run(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, std::__1::list<osgeo::proj::io::SQLValues, std::__1::allocator<osgeo::proj::io::SQLValues> > const&) in libproj.a(factory.cpp.o)
  "_sqlite3_value_double", referenced from:
      osgeo::proj::io::PROJ_SQLITE_GetValAsDouble(sqlite3_value*, bool&) in libproj.a(factory.cpp.o)
  "_sqlite3_value_int64", referenced from:
      osgeo::proj::io::PROJ_SQLITE_GetValAsDouble(sqlite3_value*, bool&) in libproj.a(factory.cpp.o)
  "_sqlite3_value_type", referenced from:
      osgeo::proj::io::PROJ_SQLITE_GetValAsDouble(sqlite3_value*, bool&) in libproj.a(factory.cpp.o)
  "_sqlite3_vfs_find", referenced from:
      osgeo::proj::io::DatabaseContext::Private::createCustomVFS() in libproj.a(factory.cpp.o)
      osgeo::proj::io::VFSClose(sqlite3_file*) in libproj.a(factory.cpp.o)
  "_sqlite3_vfs_register", referenced from:
      osgeo::proj::io::DatabaseContext::Private::createCustomVFS() in libproj.a(factory.cpp.o)
  "_sqlite3_vfs_unregister", referenced from:
      osgeo::proj::io::DatabaseContext::Private::~Private() in libproj.a(factory.cpp.o)
ld: symbol(s) not found for architecture x86_64
```

I don't have osx machine, can anyone help me?

JackBoosY commented:
/azp run

JackBoosY committed:
Merge branch 'master' into dev/jack/upgrade_proj4_6.1.1

Commit:
[libspatialite]Fix linux build.

Commit:
[libspatialite]Fix linux command.

Commit:
[libspatialite]Fix configure error.

Commit:
[proj4]fix static cmake tools path.

JackBoosY committed:
Merge branch 'master' into dev/jack/upgrade_proj4_6.1.1

Commit:
[libgeotiff]Set configure options to 1/0 to avoid the "if" comparison failure in config.cmake.
[proj4]Revert changes about namespace.
[liblas]Remove FindPROJ4.cmake and FindGeoTIFF.cmake from the source.

Commit:
Merge branch 'dev/jack/upgrade_proj4_6.1.1' of https://github.com/JackBoosY/vcpkg into dev/jack/upgrade_proj4_6.1.1

Commit:
[vtk]Fix using latest proj4 issue.

JackBoosY commented:
@dan-shaw  I can't reproduce the qt5 component build failure on my machine. Can you test them?
If it is fake, this PR should be merged.

JackBoosY committed:
Merge branch 'master' into dev/jack/upgrade_proj4_6.1.1

JackBoosY committed:
Merge branch 'master' into dev/jack/upgrade_proj4_6.1.1

Commit:
[proj4]Delete missing patch.

dan-shaw commented:
Besides the `ports/sqlite3/CONTROL` merge conflict, LGTM.

JackBoosY committed:
Merge branch 'master' into dev/jack/upgrade_proj4_6.1.1

JackBoosY committed:
Merge branch 'master' into dev/jack/upgrade_proj4_6.1.1

JackBoosY commented:
Already merged 9 times...OMG..

cbezault commented:
/azp run

dan-shaw commented:
/azp run
